xrpl/models/requests/ledger_entry.py

```diff
@@ -132,6 +133,29 @@ class Offer(BaseModel):
     """
 
 
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class Oracle(BaseModel):
+    """
+    Required fields for requesting a Price Oracle Ledger Entry, if not querying by
+    object ID.
+    """
+
+    account: str = REQUIRED  # type: ignore
+    """
+    This field is required.
+
+    :meta hide-value:
+    """
+
+    oracle_document_id: Union[str, int] = REQUIRED  # type: ignore
```

Comment:
Why use `Union[str, int]` when this field is an `int`?

Reply:
The rippled implementation allows for both a `std::uint32_t` and a `std::string`: https://github.com/XRPLF/rippled/blob/f6509495731b396ce33d96778e8df0366a8200bf/src/ripple/rpc/handlers/LedgerEntry.cpp#L625

Reply:
Here are some unit tests that make use of both the types: https://github.com/XRPLF/rippled/blob/f6509495731b396ce33d96778e8df0366a8200bf/src/test/jtx/impl/Oracle.cpp#L267

Reply:
Ah I didn't know. Great, thanks for explaining this!